src/components/list/list-item-user/ListItemUser.tsx

```diff
@@ -0,0 +1,33 @@
+import React from 'react'
+import { ListItem, ListItemText, ListItemSecondaryAction, Switch, FormControlLabel } from '@material-ui/core'
+import { User } from '../../../interfaces';
+import { useDispatch } from 'react-redux';
+import { updateUser } from '../../../store/actions';
+
+interface ComponentProps {
+    data: User;
+}
+
+const ListItemUser: React.FC<ComponentProps> = (props) => {
+    const dispatch = useDispatch()
+
+    const onChangeHandler = (e: React.ChangeEvent<HTMLInputElement>) => {
+        dispatch(updateUser(props.data.email, e.target.checked))
+    }
+
+    return <ListItem button divider>
+        <ListItemText primary={props.data.email} />
+        <ListItemSecondaryAction>
+            <FormControlLabel
+                control={<Switch
+                    checked={props.data.isAdmin === 1 ? true : false}
```

Comment:
Try to modify this row to:
checked={props.data.isAdmin} 
I think, it must work, because 0 is equal to false and 1 is equal to true in JS (but if you use == operator, not ===).
Or if this approach does not work, try to change the type of isAdmin in its initialization.